rules/transition_support.bzl

```diff
@@ -15,13 +15,17 @@ def _current_apple_platform(apple_fragment, xcode_config):
 
 def _cpu_string(platform_type, settings):
     """Generates a <platform>_<arch> string for the current target based on the given parameters."""
+
+    # If the cpu value has already been transformed to the correct value, we must not change it anymore.
+    # Otherwise, we may build for the wrong arch.
+    cpu_value = settings["//command_line_option:cpu"]
+    if (platform_type == "macos" and cpu_value.startswith("{}_".format(platform_type))) or cpu_value.startswith("{}_".format(platform_type)):
+        return cpu_value
```

Comment:
So IIUC in the old logic `ios_multi_cpus` would return early if present and this check wouldn't happen. By moving this up you're giving `cpu` a higher priority and not doing the check for `ios_multi_cpus` that would've happened in the old logic.

Can this change in behaviour be considered a breaking change?

Reply:
The original logic is broken for multi arches because 1) ios_multi_cpus  is never changed by any transition, and 2) `apple_common.multi_arch_split` may have already changed cpu based on ios_multi_cpus.

Consider this example: two frameworks ( fw1 -> fw2) are built with --ios_multi_cpus=x86_64,arm64. The outgoing edge transition of fw1 will change the cpu of the configured target fw2_arm64 to ios_arm64. In the original logic, the incoming edge transition of fw2 will always change cpu to ios_x86_64 because x86_64 is the first one of ios_multi_cpus. Then fw2_arm64 is actually built for x86_64.

This transition behavior doesn't change for a single ios_multi_cpus. Therefore, I will not say it's a breaking change.

Reply:
Awesome thx for the detailed explanation!